src/azure-cli/azure/cli/command_modules/storage/operations/blob.py

```diff
@@ -500,123 +501,77 @@ def _adjust_block_blob_size(client, blob_type, length):
-            # used to check for the preconditions as append_blob_from_path() cannot
-            client.get_blob_properties(**check_blob_args)
-        else:
-            client.create_blob(content_settings=content_settings, metadata=metadata, **check_blob_args)
-
-        append_blob_args = {
-            'container_name': container_name,
-            'blob_name': blob_name,
-            'file_path': file_path,
-            'progress_callback': progress_callback,
-            'maxsize_condition': maxsize_condition,
-            'lease_id': lease_id,
-            'timeout': timeout
-        }
-
-        if cmd.supported_api_version(min_api='2016-05-31'):
-            append_blob_args['validate_content'] = validate_content
+    if progress_callback:
+        upload_args['raw_response_hook'] = progress_callback
 
-        return client.append_blob_from_path(**append_blob_args)
+    check_blob_args = {
+        'if_modified_since': if_modified_since,
+        'if_unmodified_since': if_unmodified_since,
+        'if_match': if_match,
+        'if_none_match': if_none_match,
+    }
 
-    def upload_block_blob():
-        # increase the block size to 100MB when the block list will contain more than 50,000 blocks
-        if os.path.isfile(file_path) and os.stat(file_path).st_size > 50000 * 4 * 1024 * 1024:
-            client.MAX_BLOCK_SIZE = 100 * 1024 * 1024
-            client.MAX_SINGLE_PUT_SIZE = 256 * 1024 * 1024
+    # used to check for the preconditions as upload_append_blob() cannot
+    if blob_type == 'append':
+        if client.exists(timeout=timeout):
+            client.get_blob_properties(lease=lease_id, timeout=timeout, **check_blob_args)
 
-        create_blob_args = {
-            'container_name': container_name,
-            'blob_name': blob_name,
-            'file_path': file_path,
-            'progress_callback': progress_callback,
-            'content_settings': content_settings,
-            'metadata': metadata,
-            'max_connections': max_connections,
-            'lease_id': lease_id,
-            'if_modified_since': if_modified_since,
-            'if_unmodified_since': if_unmodified_since,
-            'if_match': if_match,
-            'if_none_match': if_none_match,
-            'timeout': timeout
+    # Because the contents of the uploaded file may be too large, it should be passed into the a stream object,
+    # upload_blob() read file data in batches to avoid OOM problems
+    try:
+        if file_path:
+            length = os.path.getsize(file_path)
+            _adjust_block_blob_size(client, blob_type, length)
+            with open(file_path, 'rb') as stream:
+                response = client.upload_blob(data=stream, length=length, metadata=metadata,
+                                              encryption_scope=encryption_scope,
+                                              **upload_args, **kwargs)
+        if data is not None:
+            _adjust_block_blob_size(client, blob_type, length)
```

Comment:
Please add a fix for `_adjust_block_blob_size()`:
https://github.com/Azure/azure-cli/blob/97fa049e35aae7fc52cf2267e4384a9a4575ee09/src/azure-cli/azure/cli/command_modules/storage/operations/blob.py#L491-L499

Line 497 should be: `if length is not None and length > 50000 * 100 * 1024 * 1024:`

This is a known issue in storage-blob-extension